src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.command.Command;
+import duke.ui.Ui;
+import duke.parser.Parser;
+import duke.tasklist.TaskList;
+import duke.storage.Storage;
+
+/**
+ * Main class for Duke.
+ */
+public class Duke {
+    private Ui ui;
+    private Parser parser;
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Initialises all necessary objects for Duke.
+     *
+     * @param filePath path to duke.txt save file
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        parser = new Parser();
+        try {
+            storage = new Storage(filePath);
+            taskList = new TaskList(storage.readDuke());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Gets a response from user and sends back the output.
+     *
+     * @param input input String from user
+     * @return output String from duke
+     */
+    public String getResponse(String input) {
+        Command command = new Command(parser.getCommand(input), input);
```

Comment:
It seems that parser.getCommand(input) outputs a Enum value of Comd. My approach here is to get Parser to parse the input and output the appropriate Command. Something seems off with my approach as I think Parser should not know about Command. But over here you have 2 input variables inside your Command constructor. Looks repeated. What's your view on this?